src/components/services-view/services-view.controller.ts

```diff
@@ -180,4 +181,35 @@ export class ServicesViewController implements angular.IController {
   private updateActiveCardStyles() {
     this.$timeout(this.resizeExpansion, 50);
   }
+
+  private setRecentlyViewed() {
+    let serviceToOrder: any = this.ctrl.selectedImageStream || this.ctrl.selectedServiceClass;
+    let recentlyViewed: any = localStorage.getItem('catalog-recently-viewed-services');
+
+    // convert from strings to objects
+    recentlyViewed = recentlyViewed ? JSON.parse(recentlyViewed) : [];
+    recentlyViewed = _.map(recentlyViewed, (item: any) => {
+      return JSON.parse(item);
+    });
+
+    // if previously viewed, remove from list
+    recentlyViewed.map((item, index, array) => {
+      if (item.resource.metadata.uid === serviceToOrder.resource.metadata.uid) {
+        array.splice(index, 1);
+      }
+    });
+
+    // limit to 3 max.
+    if (recentlyViewed.length >= 3) {
+      recentlyViewed.pop();
+    }
+
+    // convert back to strings for local strorage
+    recentlyViewed = _.map(recentlyViewed, (item: any) => {
+      return JSON.stringify(item);
```

Comment:
I'd rather just store the UID and not the whole item. Then we can pass the service classes to the project summary indexed by UID.

This could result in stale display names or icons if the service class changes. It also could result in errors if a service class was deleted after you looked at it.